release-tools/build.make

```diff
@@ -67,8 +67,11 @@ endif
 # separated by semicolon. An empty variable or empty entry (= just a
 # semicolon) builds for the default platform of the current Go
 # toolchain.
+ifdef os
+BUILD_PLATFORMS=$(os)
+else
 BUILD_PLATFORMS =
-
+endif
 # This builds each command (= the sub-directories of ./cmd) for the target platform(s)
 # defined by BUILD_PLATFORMS.
 build-%: check-go-version-go
```

Comment:
I think it's this "check-go-version-go" build rule that is problematic for Windows. We may need to make changes in csi-release-tools to decouple these steps. cc @pohly

Reply:
We can add to `build.make`:

```
CHECK_GO_VERSION ?= check-go-version-go

build-%: $(CHECK_GO_VERSION)
```

Then the top-level Makefile can disable the check with:
```
CHECK_GO_VERSION =
include release-tools/build.make
```

But as discussed elsewhere, why don't we fix the check script so that it also works on Windows? Can we assume that POSIX shell scripts work on Windows?

Reply:
Agree, I think making it work in Windows is the preferable approach. I'm not sure how much work it's going to take, so a short term workaround for now is to decouple the rule.